tower-watch/src/lib.rs

```diff
@@ -72,6 +74,32 @@ impl<T, B: Bind<T>> Service for WatchService<T, B> {
     }
 }
 
+// ==== impl Error ====
+
+impl<E> fmt::Display for Error<E>
+where
+    E: fmt::Display
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Error::WatchError(_) => f.pad("watch error"),
+            Error::Inner(e) => fmt::Display::fmt(e, f),
+        }
+    }
+}
+
+impl<E> error::Error for Error<E>
+where
+    E: error::Error,
+{
+    fn cause(&self) -> Option<&error::Error> {
+        match self {
+            Error::WatchError(_) => None,
+            Error::Inner(e) => Some(e),
```

Comment:
Hmm is the inner error the cause or should this fn proxy to the inner error’s cause?

Thoughts @seanmonstar?

Reply:
It should proxy. I've been bitten by this before. Consider things that print a cause chain:

```rust
println!("error: {}", err);

loop {
    err = err.cause();
    if let Some(err) = err {
        println!("\tCaused by: {}", err);
        err = err.cause();
    } else {
        break;
    }
}
```

Since the inner error would be printed as the root, it's confusing to then see it also as the cause.

Reply:
Changed it to proxy in 142f0fcfe1c85002999925ba8e99f46637314bd0.